src/material/datepicker/calendar-body.ts

```diff
@@ -195,23 +195,29 @@ export class MatCalendarBody implements OnChanges, OnDestroy {
     return cellNumber == this.activeCell;
   }
 
-  /** Focuses the active cell after the microtask queue is empty. */
+  /**
+   * Focuses the active cell after the microtask queue is empty.
+   *
+   * Adds a 20ms timeout to fix Voiceover losing focus when pressing PageUp/PageDown (issue #24330).
+   */
   _focusActiveCell(movePreview = true) {
-    this._ngZone.runOutsideAngular(() => {
-      this._ngZone.onStable.pipe(take(1)).subscribe(() => {
-        const activeCell: HTMLElement | null = this._elementRef.nativeElement.querySelector(
-          '.mat-calendar-body-active',
-        );
-
-        if (activeCell) {
-          if (!movePreview) {
-            this._skipNextFocus = true;
+    setTimeout(() => {
+      this._ngZone.runOutsideAngular(() => {
```

Comment:
Should the `runOutsideAngular` be outside of the `setTimeout`?

Reply:
yes, that's a good idea. I moved the timeout to be inside of the `onStable`. `The less things happening during the `timeOut` timer, hopeful the more consistent behavior we should get. It might take longer for `runOutsideAngular`, and `onStable` to happen depending on the situation.